pyteal/compiler/optimizer/optimizer.py

```diff
@@ -24,10 +25,25 @@ class OptimizeOptions:
             that have no load dependencies elsewhere.
     """
 
-    def __init__(self, *, scratch_slots: bool = False):
-        self.scratch_slots = scratch_slots
+    def __init__(self, *, scratch_slots: bool = False, frame_pointers: bool = False):
+        self.scratch_slots: Final[bool] = scratch_slots
+        self.frame_pointers: Final[bool] = frame_pointers
         self._skip_slots: Set[ScratchSlot] = set()
 
+    @classmethod
+    def maximize(cls) -> "OptimizeOptions":
+        return cls(scratch_slots=True, frame_pointers=True)
```

Comment:
Here's a potential problem with maximize: v7 and lower programs can't use it because they don't support frame pointers.

If we want this method to be useful, it should enable every optimization that's possible for every program version.

Reply:
Removing for now